litr-demo/src/main/java/com/linkedin/android/litr/demo/RecordCamera2Fragment.kt

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2022 LinkedIn Corporation
+ * All Rights Reserved.
+ *
+ * Licensed under the BSD 2-Clause License (the "License").  See License in the project root for
+ * license information.
+ *
+ * Author: Ian Bird
+ */
+package com.linkedin.android.litr.demo
+
+import android.Manifest
+import android.app.Activity
+import android.app.AlertDialog
+import android.content.Context
+import android.content.pm.PackageManager
+import android.hardware.camera2.*
+import android.os.Build
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.SurfaceHolder
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.RequiresApi
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import com.linkedin.android.litr.MediaTransformer
+import com.linkedin.android.litr.demo.data.TargetMedia
+import com.linkedin.android.litr.demo.data.TransformationPresenter
+import com.linkedin.android.litr.demo.data.TransformationState
+import com.linkedin.android.litr.demo.databinding.FragmentCamera2RecordBinding
+import com.linkedin.android.litr.io.AudioRecordMediaSource
+import com.linkedin.android.litr.io.Camera2MediaSource
+import com.linkedin.android.litr.utils.TransformationUtil
+import java.io.File
+import java.util.*
+
+private const val TAG = "RecordCamera2Fragment"
+
+private const val REQUEST_AUDIO_AND_CAMERA_PERMISSION = 27
+
+private const val DEFAULT_CAMERA_FPS = 30
+private const val DEFAULT_TARGET_BITRATE = 5_000_000 // 5Mbps
+private const val DEFAULT_RECORD_WIDTH = 1280
```

Comment:
Yeah, makes sense. I'll have a think, and potentially push another update. Maybe, while we have the aspect ratio requirement, we could do something like a "maximum resolution" that we compute our target resolution within. For this example case, it would be like passing in a max resolution of "1280x1280" which is basically the max bounding box?